src/mlpack/methods/ann/ffn.hpp

```diff
@@ -324,6 +325,14 @@ class FFN
   //! Modify the matrix of data points (predictors).
   arma::mat& Predictors() { return predictors; }
 
+  /**
+   * Check wether the input shape is consistent with the layer requirements.
+   * 
+   * @param inputShape shape of the input
+   * @param functionName function that checks the input size
+   */  
+  void CheckInputShape(size_t inputShape, std::string functionName);
```

Comment:
```suggestion
  void CheckInputShape(const size_t inputShape, const std::string& functionName);
```

To avoid an extra copy.